applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_linear_elastic_plane_strain_2D_law.cpp

```diff
@@ -0,0 +1,201 @@
+
+    Vector strain;
+    parameters.SetStrainVector(strain);
+
+    rConstitutiveLaw.CalculateMaterialResponsePK2(parameters);
+
+    return strain;
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawReturnsExpectedStress, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+
+    const auto stress = CalculateStress(law);
+
+    Vector expected_stress{4};
+    expected_stress <<= 2.5e+07, 2.5e+07, 2.5e+07, 3.84615e+06;
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(expected_stress, stress, 1e-3);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawReturnsExpectedStress_WhenOnlyDiagonalEntriesAreConsidered,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+    law.SetConsiderDiagonalEntriesOnlyAndNoShear(true);
+
+    const auto stress = CalculateStress(law);
+
+    Vector expected_stress{4};
+    expected_stress <<= 1.34615e+07, 1.34615e+07, 1.34615e+07, 0;
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(expected_stress, stress, 1e-3);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawReturnsExpectedStress_WithInitialStressAndStrain,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+
+    ConstitutiveLaw::Parameters parameters;
+    auto                        initial_strain = Vector{ScalarVector{4, 0.5}};
+    parameters.SetStrainVector(initial_strain);
+    auto initial_stress = Vector{ScalarVector{4, 1e6}};
+    parameters.SetStressVector(initial_stress);
+    law.InitializeMaterialResponseCauchy(parameters);
+
+    const auto stress = CalculateStress(law);
+
+    Vector expected_stress{4};
+    expected_stress <<= 1.35e+07, 1.35e+07, 1.35e+07, 2.92308e+06;
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(expected_stress, stress, 1e-3);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawReturnsExpectedStress_AfterFinalizeMaterialResponse,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+
+    ConstitutiveLaw::Parameters initial_parameters;
+    auto                        initial_strain = Vector{ScalarVector{4, 0.5}};
+    initial_parameters.SetStrainVector(initial_strain);
+    auto initial_stress = Vector{ScalarVector{4, 1e6}};
+    initial_parameters.SetStressVector(initial_stress);
+    law.InitializeMaterialResponseCauchy(initial_parameters);
+
+    auto stress = CalculateStress(law);
+
+    ConstitutiveLaw::Parameters final_parameters;
+    auto                        final_strain = Vector{ScalarVector{4, 1.3}};
+    final_parameters.SetStrainVector(final_strain);
+    law.FinalizeMaterialResponseCauchy(final_parameters);
+    stress = CalculateStress(law);
+
+    Vector expected_stress{4};
+    expected_stress <<= 6e+06, 6e+06, 6e+06, 1.76923e+06;
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(expected_stress, stress, 1e-3);
+}
```

Comment:
I treated these tests as regression tests, i.e. before I did the refactoring, I checked what the output was and set that as the expected result. I didn't do an analytical comparison